Yoast/wpseo-woocommerce
# Bring back product metatags

## TODO

* Write tests

## Context


* We should...
  1. ...still output our metatags as before
  2. ...not trigger our own deprecation notices
  3. ...show deprecation notices when the functions are still called by external parties

## Summary

This PR can be summarized in the following changelog entry:

* Adds a WooCommerce presenter to bring back the product metatags that relied on the (deprecated) `wpseo_opengraph` action.

## Relevant technical choices:

* I'm not sure about...
  * ...the location of the deprecation of the `Yoast\WP\Woocommerce\opengraph` action. I put it there because the old location (`product_enhancement`) is deprecated too.
  * ...the removal of the action in the old `product_enhancement` method.
* I chose not to update too much, if you spot some easy fixes, feel free to apply them 😉 

## Test instructions

This PR can be tested by following these steps:

* Install WooCommerce and WooCommerce Brands and set up a product with a brand, price, availability and so on.
* Compare the metatag output before this branch and after this branch, there should be:
  - product:brand
  - product:price:amount
  - product:price:currency
  - og:availability
  - product:availability
  - product:retailer_item_id
  - product:condition
* The metatag of `product:availability` when in stock is changed compared to what it was before. From `in stock` to `instock`. As this got noted as being wrong by Jono, see: https://github.com/Yoast/wpseo-woocommerce/issues/555 for future changes.
* You could manually run the actions that are now deprecated to see the notices.

Fixes #566 
Fixes https://github.com/Yoast/bugreports/issues/950
